dipy/reconst/dki.py

```diff
@@ -0,0 +1,1634 @@
+    d1mach[4]=np.log(2.0)
+    errtol = (d1mach[2]/3.0)**(1.0/6.0)
+    lolim  = 2.0/(d1mach[1])**(2.0/3.0)
+    tuplim = d1mach[0]**(1.0/3.0)
+    tuplim = (0.10*errtol)**(1.0/3.0)/tuplim
+    uplim  = tuplim**2.0
+    c1 = 3.0/14.0
+    c2 = 1.0/6.0
+    c3 = 9.0/22.0
+    c4 = 3.0/26.0
+
+    xn = x.copy()
+    yn = y.copy()
+    zn = z.copy()
+ 
+    mu = (xn+yn+zn)/3.0
+    xndev = 2.0 - (mu+xn)/mu
+    yndev = 2.0 - (mu+yn)/mu
+    zndev = 2.0 - (mu+zn)/mu
+    epslon = np.max([np.abs(xndev),np.abs(yndev),np.abs(zndev)])
+    while (epslon >= errtol):
+       xnroot = np.sqrt(xn)
+       ynroot = np.sqrt(yn)
+       znroot = np.sqrt(zn)
+       lamda = xnroot*(ynroot+znroot) + ynroot*znroot
+       xn = (xn+lamda)*0.250
+       yn = (yn+lamda)*0.250
+       zn = (zn+lamda)*0.250
+       mu = (xn+yn+zn)/3.0
+       xndev = 2.0 - (mu+xn)/mu
+       yndev = 2.0 - (mu+yn)/mu
+       zndev = 2.0 - (mu+zn)/mu
+       epslon = np.max([np.abs(xndev),np.abs(yndev),np.abs(zndev)])
+
+    e2 = xndev*yndev - zndev*zndev
+    e3 = xndev*yndev*zndev
+    s  = 1.0 + (c1*e2-0.10-c2*e3)*e2 + c3*e3
+    drf = s/np.sqrt(mu)
+    return drf
+
+
+def C2222(a,b,c):
+    """
+    WIP
+    """
+    Carray=np.zeros(a.shape)
+    abc= np.array((a, b, c))
+    indexesxcond1=np.logical_and.reduce(abc>0)
+    if np.sum(indexesxcond1)!=0:
+      Carray[indexesxcond1]=((a[indexesxcond1]+2.*b[indexesxcond1])**2/(24.*b[indexesxcond1]**2))
+    
+    indexesxcond2=np.logical_and(np.logical_and.reduce(abc>0),(b!=c))
+    if np.sum(indexesxcond2)!=0:
+      Carray[indexesxcond2]=(((a[indexesxcond2]+b[indexesxcond2]+c[indexesxcond2])**2/((18.)*(b[indexesxcond2])*(b[indexesxcond2]-c[indexesxcond2])**2))*(2.*b[indexesxcond2]+((c[indexesxcond2]**2-3.*b[indexesxcond2]*c[indexesxcond2])/(np.sqrt(b[indexesxcond2]*c[indexesxcond2])))))
+
+  ### the following condition has to be checked ###
+    indexesxcond3=np.logical_or.reduce(abc<=0)
+    Carray[indexesxcond3]=0   
+    return Carray  
+
+
+def C2233(a,b,c):
+    """
+    WIP
+    """
+    Carray=np.zeros(a.shape)
+    abc= np.array((a, b, c))
+    
+    indexesxcond1=np.logical_and.reduce(abc>0)
+    if np.sum(indexesxcond1)!=0:
+      Carray[indexesxcond1]=((a[indexesxcond1]+2.*b[indexesxcond1])**2/(12.*b[indexesxcond1]**2))
+      
+    indexesxcond2=np.logical_and(np.logical_and.reduce(abc>0),(b!=c))
+    if np.sum(indexesxcond2)!=0:
+      Carray[indexesxcond2]=(((a[indexesxcond2]+b[indexesxcond2]+c[indexesxcond2])**2/((3.)*(b[indexesxcond2]-c[indexesxcond2])**2))*(((b[indexesxcond2]+c[indexesxcond2])/(np.sqrt(b[indexesxcond2]*c[indexesxcond2])))-2.))
+
+  ### the following condition has to be checked ###
+    indexesxcond3=np.logical_or.reduce(abc<=0)
+    Carray[indexesxcond3]=0   
+    return Carray  
+
+
+def F1m(a,b,c):
+    """
+    Helper function that computes function $F_1$ which is required to compute
+    the analytical solution of the Mean kurtosis.
+    
+    Parameters
+    ----------
+    a : ndarray (...)
```

Comment:
So, what do you mean here that your variables can have any dimensions? Please clarify.